ext/pgsql/pgsql.c

```diff
@@ -1698,6 +1698,14 @@ PHP_FUNCTION(pg_fetch_result)
 	int pgsql_row;
 
 	if (ZEND_NUM_ARGS() == 2) {
+		if (zend_string_equals_literal(EX(func)->common.function_name, "pg_fetch_result")) {
+			zend_error(E_DEPRECATED, "Calling pg_fetch_result() with 2 arguments is deprecated, "
+				"use the 3-parameter signature with a null $row parameter instead");
+			if (UNEXPECTED(EG(exception))) {
+				RETURN_THROWS();
+			}
+		}
+
 		ZEND_PARSE_PARAMETERS_START(2, 2)
```

Comment:
Is this because the implementations are aliased? Wouldn't it be better to just define two functions, and call a common implementation function?

Reply:
Yes, it's because there's the `pg_result()` alias function. It's also possible to separate the implementations into two `PHP_FUNCTION`s which call the same C function, but I wanted to do the least amount of code changes: as we are pretty late in the PHP 8.x series, most probably this piece of hack wouldn't be with us for too long.

Reply:
I m fine keeping this PR as is, the pgsql changes itself looks fine by me.